File: C_Programming/Practise_Programs/PP_040.c
/*
==============================================================================
 Program     : Display Numbers from N to 1

 Description : Reads an integer from the user and displays all numbers from
               the specified value down to 1.

 Author      : Shubham Gadhe
==============================================================================
*/

#include<stdio.h>

/*
==============================================================================
 Function    : Display

 Description : Displays all numbers from the specified value down to 1.

 Parameters  : iNo - Starting value of the sequence.

 Returns     : None

 Complexity  : Time  : O(n)
               Space : O(1)
==============================================================================
*/

void Display(int iNo)
{
    int iCnt = 0;

    // Display numbers from the specified value down to 1.
    for(iCnt = iNo; iCnt >= 1; iCnt--)
    {
        printf("%d\t", iCnt);
    }
    
    printf("\n");

}

int main()
{
    int iValue = 0;

    printf("Enter number : \n");
    scanf("%d", &iValue);

    Display(iValue);

    return 0;
}

/*
==============================================================================
 Sample Input
------------------------------------------------------------------------------
 5

 Sample Output
------------------------------------------------------------------------------
 5    4    3    2    1
==============================================================================
*/

File: Java_Programming/Practise_Programs/PP_040.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_040
//
//  Description       : This program accepts a number from the user
//                      and displays numbers from 1 to the given
//                      number, with '*' printed after each number.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 06/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.*;

class PP_040
{
    //////////////////////////////////////////////////////////////////
    //
    //  Function Name    : Display()
    //
    //  Description      : It is used to display numbers
    //                     from 1 to the specified number,
    //                     followed by '*' after each number.
    //
    //  Author           : Shubham Somanath Gadhe
    //  Date             : 06/08/2026
    //
    //////////////////////////////////////////////////////////////////

    public static void Display(int iNo)
    {
        int iCnt = 0;

        for(iCnt = 1; iCnt <= iNo; iCnt++)
        {
            System.out.print(iCnt + "\t" + "*" + "\t");
        }

        System.out.println();
    }

    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;

        System.out.println("Enter the number of elements : ");
        iValue = sobj.nextInt();

        Display(iValue);
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input  :
//      5
//
//  Sample Output :
//      1    *    2    *    3    *    4    *    5    *
//
//////////////////////////////////////////////////////////////////